bot/formatter.py

```diff
@@ -0,0 +1,96 @@
+# coding=utf-8
+
+"""
+Credit to Rapptz's script used as an example:
+https://github.com/Rapptz/discord.py/blob/rewrite/discord/ext/commands/formatter.py
+Which falls under The MIT License.
+"""
+
+import itertools
+from inspect import formatargspec, getfullargspec
+
+from discord.ext.commands import Command, HelpFormatter, Paginator
+
+from bot.constants import HELP_PREFIX
+
+
+class Formatter(HelpFormatter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    # basically the same function but changed:
+    # - to make the helptext appear as a comment
+    # - to change the indentation to the PEP8 standard: 4 spaces
+    def _add_subcommands_to_page(self, max_width, commands):
+        for name, command in commands:
+            if name in command.aliases:
+                # skip aliases
+                continue
+
+            entry = "    {0:<{width}} # {1}".format(name, command.short_doc, width=max_width)
+            shortened = self.shorten(entry)
+            self._paginator.add_line(shortened)
+
+    async def format(self):
+        self._paginator = Paginator(prefix="```py")
+
+        if isinstance(self.command, Command):
+            # help on a specific command to look like an async function
+
+            # example:
+            # async def <command>(ctx, <args>):
+            #     """
+            #     <help text>
+            #     """
+            #     await do_<command>(ctx, <args>)
+
+            # strip the command of bot. and ()
```

Comment:
this block (from here and until the return) generally feels too dense. PEP8 allows blank lines used sparingly to split code into logical sections, and this can be a real gift for good readability. personally I try to write comments that encapsulate the actions of at least 3-4 lines of code and then put blank lines before the comments.

I also try to make sure similar things are grouped together, and that similar actions are handled consistently. the comments should also apply the same tense (don't mix present tense and past tense). applying that philosophy to this section, I'd do something like this;

```py
# strip the command of bot. and ()
stripped_command = self.command.name.replace(HELP_PREFIX, "").replace("()", "")

# get the args using the handy inspect module
argspec = getfullargspec(self.command.callback)
arguments = formatargspec(*argspec)
args_no_type_hints = ", ".join(argspec[0])

# remove self from the args
arguments = arguments.replace("self, ", "")
args_no_type_hints = args_no_type_hints.replace("self, ", "")

# prepare the different sections of the help output, and add them to the paginator
definition = f"async def {stripped_command}{arguments}:"
docstring = f"    \"\"\"\n    {self.command.help}\n    \"\"\""
invocation = f"    await do_{stripped_command}({args_no_type_hints})"
self._paginator.add_line(definition)
self._paginator.add_line(docstring)
self._paginator.add_line(invocation)

return self._paginator.pages
```